controllers/runner_controller.go

```diff
@@ -775,6 +775,10 @@ func newRunnerPod(template corev1.Pod, runnerSpec v1alpha1.RunnerConfig, default
 			Name:  "RUNNER_GROUP",
 			Value: runnerSpec.Group,
 		},
+		{
+			Name:  "DOCKER_ENABLED",
+			Value: fmt.Sprintf("%v", dockerEnabled),
```

Comment:
AFAIK `dockerEnabled` doesn't turn `true` for `dockerdWithinRunnerContainer` where you'd also want it to wait for dockerd startup.

How about:

```suggestion
			Value: fmt.Sprintf("%v", dockerEnabled || dockerdInRunner),
```